linux/LinuxProcessList.c

```diff
@@ -1008,33 +1008,39 @@ static inline void LinuxProcessList_scanZramInfo(LinuxProcessList* this){
    unsigned long long int usedZramComp = 0;
    unsigned long long int usedZramOrig = 0;
 
-   char mm_stat[32];
-   char disksize[32];
+   char mm_stat[34];
+   char disksize[34];
 
-   FILE* file;
    unsigned int i = 0;
-   snprintf(mm_stat,32,"/sys/block/zram%u/mm_stat",i);
-   snprintf(disksize,32,"/sys/block/zram%u/disksize",i);
-   while((file = fopen(disksize,"r")) != NULL){
-      unsigned long long int size;
-      fscanf(file,"%llu\n",&size);
-      totalZram += size/1024;
-      fclose(file);
-
-      file = fopen(mm_stat,"r");
-      unsigned long long int orig_data_size;
-      unsigned long long int compr_data_size;
+   do{
+      snprintf(mm_stat,34,"/sys/block/zram%u/mm_stat",i);
+      snprintf(disksize,34,"/sys/block/zram%u/disksize",i);
+      i++;
+      FILE* disksize_file = fopen(disksize,"r");
+      FILE* mm_stat_file = fopen(mm_stat,"r");
+      if(disksize == NULL || mm_stat_file == NULL){
```

Comment:
Leaks one `FILE*` if only one of the files could be opened …

Reply:
"if(disksize == NULL || mm_stat_file == NULL)" already fixed. now it's "if(disksize_file == NULL || mm_stat_file == NULL)"

Reply:
Still, `break`ing if one of them is `NULL` does not guarantee the other one is too.

```diff
 if(disksize_file == NULL || mm_stat_file == NULL) {
+   if(disksize_file != NULL) {
+      fclose(disksize_file);
+   }
+   if(mm_stat_file != NULL) {
+      fclose(mm_stat_file);
+   }
    break;
 }
```